ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcRecordUpdater.java

```diff
@@ -579,6 +579,7 @@ public void close(boolean abort) throws IOException {
 
             OrcFile.WriterOptions wo = OrcFile.writerOptions(this.options.getConfiguration())
                 .inspector(rowInspector)
+                .setSchema(createEventSchemaFromTableProperties(this.options.getTableProperties()))
```

Comment:
Why is this needed after this patch?

Reply:
I simplified this part of the code: reuse the existing writerOptions instance which contains the correct schema.